### ISSUE TITLE
Events: Search filters - remove countries without events

### ISSUE THREAD

**Issue by pkevan** [WordPress/wordcamp.org#1173]
Having a list of every country in the world is unnecessary as a search filter - it would be more appropriate to list the countries where events are going to take place, than having a huge list to filter where they might not return any results.

**Comment by pkevan** [comment]
![Screenshot 2023-12-11 at 08 57 19](https://github.com/WordPress/wordcamp.org/assets/2290623/5696bb92-e957-4224-8c74-165d37730826)

Example: all countries which appear in the list currently have no events.

**Comment by iandunn** [comment]
Right now it's using `wcorg_get_countries()`. I think we'd have to change it to query `wporg_events`, and get the distinct country codes where there's at least 1 upcoming event. Then we'd need to JOIN to one of the Geonames tables to get the country name.

We'd also need to switch that to query past events when looking at the Past Events template.

That might introduce some problems w/ country names that CLDR solves, but that's probably a smaller issue that what's being fixed.

I wonder if the need for this will be removed once [we add searching](https://github.com/WordPress/wporg-mu-plugins/issues/541)? That should make it easy for folks to find what they're looking for, and at that point the need for this seems minimal to me. What do you think?

**Comment by pkevan** [comment]
Yeah, I looked into it a bit yesterday, and found it was quite complicated - I was hoping that the stat `108 countries` on the homepage was a cached query somewhere that would help this, but turns out it's manual. The other option I thought about was through: https://central.wordcamp.org/reports/meetup-groups-report/?report-year=2023&period=all&action=Show+results which returns countries.

It could well be solved with searching, since I wonder who really uses filters these days.
